doc/design/refactor_discuess/net_design.md

```diff
@@ -0,0 +1,65 @@
+# Network Design
+
+`Network` is the container and controller of a set of operators in a network, users can use `Network.addOp` to add operators into a network, 
+and use `Network.runOps` to run all the operators in the network.
+
+The `Network` will
+
+- manage all the operators contained in the network.
+- not own any `Variable`.
+
+# API
+
+To make the `Network` extendibe, a base class is defined like this
+
+```c++
+// The minimum a network should be implemented.
+class BaseNetwork {
```

Comment:
Net seems wired without a prefix like "SimpleNet" or "DAGNet" and I will name it "NetworkBase"